src/main/java/org/truffleruby/language/RubyNode.java

```diff
@@ -36,8 +42,38 @@ public Object isDefined(VirtualFrame frame) {
         return coreStrings().EXPRESSION.createInstance();
     }
 
+    // Instrumentation
+
+    public WrapperNode createWrapper(ProbeNode probe) {
+        return new RubyNodeWrapper(this, probe);
+    }
+
+    public boolean isInstrumentable() {
+        return hasSource();
+    }
+
     // Boundaries
 
+    public boolean hasTag(Class<? extends Tag> tag) {
+        if (tag == TraceManager.CallTag.class || tag == StandardTags.CallTag.class) {
+            return isCall();
+        }
+
+        if (tag == StandardTags.StatementTag.class) {
+            return isNewLine();
+        }
+
+        if (tag == TraceManager.LineTag.class) {
+            return isNewLine() || isCoverageLine();
+        }
+
+        if (tag == StandardTags.RootTag.class) {
+            return isRoot();
+        }
+
+        return false;
+    }
```

Comment:
Is it intentional to move hasTag() here instead of in RubyBaseNode?

Reply:
Yes. `RubyBaseNode` previously overrode `isTaggedWith()` on `Node` but did not mark itself as instrumentable. `hasTag()` is now defined on `InstrumentableNode`, and since `RubyNode` is the most basic node type we do mark as instrumentable that is the class I chose to implement `InstrumentableNode` on, and so it made sense to move the method to there.

Reply:
Makes sense.